sys/arm64/arm64/machdep.c

```diff
@@ -396,6 +396,9 @@ init_proc0(vm_pointer_t kstack)
 /*
  * Get an address to be used to write to kernel data that may be mapped
  * read-only, e.g. to patch kernel code.
+ *
+ * XXX-AM: This should probably receive a size argument so that we can
+ * set bounds on the resulting capability.
```

Comment:
All the comments in this function say it only operates on a page, and given it accepts a virtual address and there is no guarantee that contiguous virtual pages are physically contiguous, the code is assuming that (it doesn't do a loop fetching multiple pages for example).  Given that, this function can use PHYS_TO_DMAP_PAGE.

Reply:
I double checked, there are only two users of this (at leas explicitly) and they both are using it for small integer-sized values.

Reply:
Moved to PHYS_TO_DMAP_PAGE